specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json

```diff
@@ -4391,6 +4436,41 @@
         },
       "description": "A skill for merging two or more strings into a single unified string, with an optional user-defined delimiter separating each component part."
     },
+    "EntityRecognitionSkill": {
+      "x-ms-discriminator-value": "#Microsoft.Skills.Text.EntityRecognitionSkill",
+      "allOf": [{
+        "$ref": "#/definitions/Skill"
+      }],
+      "properties": {
+        "categories": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EntityCategory",
+            "x-nullable": false
+          },
+          "description": "A list of entity categories that should be extracted."
+        },
+        "defaultLanguageCode": {
+          "$ref": "#/definitions/EntityRecognitionSkillLanguage",
+          "description": "A value indicating which language code to use. Default is en."
+        },
+        "minimumPrecision": {
+          "type": "number",
+          "format": "double",
+          "x-nullable": true,
+          "description": "Unused at the moment. Reserved for future use."
```

Comment:
Maybe you can provide guidance here... the reason for including this here is to provide the most seamless transition from NamedEntityRecognitionSkill (NER) to EntityRecognitionSkill (entity)...

NERSkill filters results based on this, but the entity skill doesn't *at the moment*
This could change in the future, when that functionality is available and therefore is part of the spec... I'd like to leave it as is, but let me know if the reasoning is not sound

Reply:
Of course I'll update the description

Reply:
Again, the property was added to help ease transition from NERSkill to the Entity skill... The user is only required at most times to change the @odata.type property...

However, I do see the merit of your point in the swagger spec/sdk generation path.
With the compilers assistance, when a user migrates from nerskill to entity skill, it should be easy enough for them remove the parameter without having to make an API. So I'll remove it

---
In reply to: [254071834](https://github.com/Azure/azure-rest-api-specs/pull/5145#discussion_r254071834) [](ancestors = 254071834)